integration_tests/tests/cucumber.rs

```diff
@@ -400,24 +416,44 @@ async fn prune_node_connected_to_base_node(
 }
 
 #[when(expr = "wallet {word} detects all transactions as Mined_Confirmed")]
-async fn wallect_detects_all_txs_as_mined_confirmed(world: &mut TariWorld, wallet_name: String) {
-    let mut client = create_wallet_client(world, wallet_name).await.unwrap();
-    let wallet_identity = client.identify(GetIdentityRequest {}).await.unwrap().into_inner();
-    let wallet_pubkey = wallet_identity.public_key.to_hex();
-    let tx_ids = world.wallet_tx_ids.get(&wallet_pubkey).unwrap();
+async fn wallet_detects_all_txs_as_mined_confirmed(world: &mut TariWorld, wallet_name: String) {
+    let mut client = create_wallet_client(world, wallet_name.clone()).await.unwrap();
+    let wallet_address = client
+        .get_address(Empty {})
+        .await
+        .unwrap()
+        .into_inner()
+        .address
+        .to_hex();
+    let tx_ids = world.wallet_tx_ids.get(&wallet_address).unwrap();
 
     let num_retries = 100;
 
     for tx_id in tx_ids {
         println!("waiting for tx with tx_id = {} to be mined_confirmed", tx_id);
-        'inner: for _ in 0..num_retries {
+        for retry in 0..=num_retries {
```

Comment:
nit: Technically this counts higher than the num of retries by 1

Reply:
The idea is to have a failure check `retry`, but can make it consistent with the other methods, as well.